### PR TITLE
feat(templates): update React InstantSearch template to Satellite

### PR DESCRIPTION
**Pull request by francoischalifour**
## Description

This updates the React InstantSearch template to the new Satellite theme.

## Changes

- Use React Scripts v3
- Use InstantSearch Satellite theme
- Use Tailwind
- Use Algoliasearch v4
- Better responsiveness (no filters panel yet because it added too much code in the template)

## How to try it

- Fetch this branch and run `yarn start /tmp/cisa-react-satellite` (choose React InstantSearch template)
- [Check on CodeSandbox](https://codesandbox.io/s/cq78p)

### CODE REVIEW

**Comment by Haroenv** [comment]
Is this using an old version of the satellite theme? The input still seems to overflow on small widths

**Comment by Haroenv** [comment]
This is more of a general note, but should we `restrictSearchableAttributes={['name', 'description']}`, since we just show those? That might be confusing for people with another index though

**Comment by francoischalifour** [comment]
@Haroenv This is the latest version of the theme. The results container overflows on xsmall devices so that the filters panel stay usable. As explained in the post, I didn't implement the filters panel here (yet) because it adds some code.

**Comment by Haroenv** [comment]
I was more talking about the fact that something seems to have a hardcoded size, since it overflows, not per se about a panel :)

**Comment by eunjae-lee** [comment]
If I understand correctly, Haroen is talking about Pagination component being too wide, breaking the responsive layout.
We might have `<Pagination padding={2} />`..?

**Comment by francoischalifour** [comment]
I reduced the `Pagination` padding in 2dba4c2283048c1f993d238c7cac1bf663a3f17a.

File: src/templates/React InstantSearch/package.json [hunk 1]
```diff
@@ -6,29 +6,40 @@
     "start": "react-scripts start",
     "build": "react-scripts build",
     "lint": "eslint .",
-    "lint:fix": "npm run lint -- --fix"
+    "build:styles": "tailwind build src/index.css -o public/tailwind.css",
```
Comment (Haroenv) [inline src/templates/React InstantSearch/package.json]: does this mean the styles don't watch? and you need to run this every time you change some css?

Comment (francoischalifour) [reply src/templates/React InstantSearch/package.json]: No, it only builds Tailwind utilities (`text-center`, etc.) when you run `yarn start` or `yarn build`. You can see the result in the sandbox and fetch the branch to try it locally.

**Comment by francoischalifour** [comment]
@Haroenv It's not really that it helped me, it's that it'll be handy to quickly prototype in the future for support requests. If ever it weighs us down, we can remove it.

**Comment by eunjae-lee** [comment]
> I understand why you did this, but I feel like the tailwind could confuse us in the future. For now I think it's clear enough that it has helped you, that we can keep it

Can you rephrase your concern? Are you confused about the usage of tailwind? Or, worried about its bundle size? Or, worried if users will get confused about it?

**Comment by Haroenv** [comment]
My concern is that it isn't regular css, and if people will copy only the jsx / css without build process, it will look very different since it won't have the tailwind classes

**Comment by francoischalifour** [comment]
@Haroenv It _is_ regular CSS ([see sandbox](https://codesandbox.io/s/cq78p)). We just generate the CSS file on `start`/`build` for convenience so that it gets updated whenever the `tailwindcss` dependency is updated.

**Comment by Haroenv** [comment]
I approved, so it's fine for me, just saying that 

```
@tailwind utilities;
```

requires preprocessing. If you don't run the prebuild or change tailwind stuff without rebuilding it won't work.

**Comment by matthew1809** [comment]
As this has an approver, is there a reason it hasn't yet been merged @francoischalifour ?

**Comment by francoischalifour** [comment]
@matthew1809 I wanted to wait for all our sandboxes to migrate to this theme to merge them approximately at the same time.

**Comment by matthew1809** [comment]
Hi @francoischalifour are we in a place to test this now?

**Comment by francoischalifour** [comment]
We first need to update all the dependencies and then apply the same theme to the other templates.

[From our discussion on Slack](https://algolia.slack.com/archives/C01BMR5AWSX/p1605515729002000) I understood that your team would like to be involved and take this work over as I won't be able to. We  can sync on the next steps if I got this right.
